tests/test_interface.rb

```diff
@@ -556,28 +555,26 @@ def test_interface_duplex_change
     interface.duplex = 'auto'
     assert_equal('auto', interface.duplex)
     interface_ethernet_default(interfaces_id[0])
-  rescue RuntimeError => e
+  rescue Cisco::CliError => e
     speed_change_disallowed(e.message)
   end
 
   def test_interface_duplex_invalid
-    skip('Duplex changes not supported on IOS XR') if platform == :ios_xr
     interface = Interface.new(interfaces[0])
     interface.speed = 1000
     assert_raises(RuntimeError) { interface.duplex = 'hello' }
     interface_ethernet_default(interfaces_id[0])
-  rescue RuntimeError => e
+  rescue Cisco::CliError => e
     speed_change_disallowed(e.message)
   end
 
   def test_interface_duplex_valid
-    skip('Duplex changes not supported on IOS XR') if platform == :ios_xr
     interface = Interface.new(interfaces[0])
     interface.speed = 1000
     interface.duplex = 'full'
     assert_equal('full', interface.duplex)
     interface_ethernet_default(interfaces_id[0])
-  rescue RuntimeError => e
+  rescue Cisco::CliError => e
     speed_change_disallowed(e.message)
```

Comment:
We should merge develop into feature/grpc after this PR is merged, so we can pull in #80